subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/internal/CheckstyleInvoker.groovy

```diff
@@ -21,114 +21,117 @@ import org.gradle.api.GradleException
-                        throw new InvalidUserDataException("Cannot add config_loc to checkstyle.configProperties. Please configure the configDirectory on the checkstyle task instead.")
-                    }
-                    // Use configDir for config_loc
-                    property(key: CONFIG_LOC_PROPERTY, value: configDir.toString())
-                }
+        ant.checkstyle(config: config.asFile, failOnViolation: false,
+            maxErrors: maxErrors, maxWarnings: maxWarnings, failureProperty: FAILURE_PROPERTY_NAME) {
+
+            source.addToAntBuilder(ant, 'fileset', FileCollection.AntType.FileSet)
+
+            if (showViolations) {
+                formatter(type: 'plain', useFile: false)
             }
 
-            if (reports.html.getRequired().get()) {
-                def stylesheet = reports.html.stylesheet ? reports.html.stylesheet.asString() :
-                    Checkstyle.getClassLoader().getResourceAsStream('checkstyle-noframes-sorted.xsl').text
-                ant.xslt(in: xmlDestination, out: reports.html.outputLocation.asFile.get()) {
-                    style {
-                        string(value: stylesheet)
-                    }
-                }
+            if (isXmlRequired || isHtmlRequired) {
+                formatter(type: 'xml', toFile: xmlOutputLocation)
             }
 
-            if (isHtmlReportEnabledOnly(reports)) {
-                GFileUtils.deleteQuietly(xmlDestination)
+            configProperties.each { key, value ->
+                property(key: key, value: value.toString())
             }
 
-            if (ant.project.properties[FAILURE_PROPERTY_NAME] && !ignoreFailures) {
-                throw new GradleException(getMessage(reports, parseCheckstyleXml(reports)))
-            } else {
-                def reportXml = parseCheckstyleXml(reports)
-                if (violationsExist(reportXml)) {
-                    logger.warn(getMessage(reports, reportXml))
+            if (configDir) {
+                // User provided their own config_loc
+                def userProvidedConfigLoc = configProperties[CONFIG_LOC_PROPERTY]
+                if (userProvidedConfigLoc) {
+                    throw new InvalidUserDataException("Cannot add config_loc to checkstyle.configProperties. Please configure the configDirectory on the checkstyle task instead.")
                 }
+                // Use configDir for config_loc
+                property(key: CONFIG_LOC_PROPERTY, value: configDir.toString())
             }
         }
-    }
 
-    private static boolean violationsExist(Node reportXml) {
-        return reportXml != null && getErrorFileCount(reportXml) > 0
-    }
+        if (isHtmlRequired) {
+            def stylesheet = stylesheetString.isPresent()
+                ? stylesheetString.get()
+                : Checkstyle.getClassLoader().getResourceAsStream('checkstyle-noframes-sorted.xsl').text
+            ant.xslt(in: xmlOutputLocation, out: htmlOutputLocation) {
```

Comment:
Something seems to be different with this change... I'm seeing this output all the time with this change:
> [ant:xslt] Warning: org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser: Property 'http://javax.xml.XMLConstants/property/accessExternalDTD' is not recognized
> [ant:xslt] Warning: org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser: Property 'http://www.oracle.com/xml/jaxp/properties/entityExpansionLimit' is not recognized.

https://ge.gradle.org/s/dlizegkowms2c/console-log#L615

This doesn't happen on current master.

Reply:
Hmm, good catch. I will check

Reply:
It seems that the culprit is having `xerces:xercesImpl`on the classpath. It's brought in by the `nekohtml` dependency. I removed both dependencies. It doesn't seem we need them, since `xerces` seems to be integrated into the JDK.

Any thoughts on that?

Reply:
I see now why we used nekohtml in some places... Since default SaxParser is very strict and `html` is not that strict. Hmm...